endpoints/openrtb2/video_auction.go

```diff
@@ -242,14 +242,23 @@ func cleanupVideoBidRequest(videoReq *openrtb_ext.BidRequestVideo, podErrors []P
 
 func handleError(labels pbsmetrics.Labels, w http.ResponseWriter, errL []error, ao analytics.AuctionObject) {
 	labels.RequestStatus = pbsmetrics.RequestStatusErr
-	w.WriteHeader(http.StatusInternalServerError)
 	var errors string
+	var foundBlacklisted bool = false
 	for _, er := range errL {
-		errors = fmt.Sprintf("%s %s", errors, er.Error())
+		if errortypes.DecodeError(er) == errortypes.BlacklistedAppCode {
+			w.WriteHeader(http.StatusServiceUnavailable)
+			ao.Status = http.StatusServiceUnavailable
+			errors = fmt.Sprintf("%s %s", er.Error(), errors)
+		} else {
+			errors = fmt.Sprintf("%s %s", errors, er.Error())
+		}
+	}
+	if !foundBlacklisted {
```

Comment:
You never set the `foundBlacklisted` anywhere in the above code, so this block will always get executed. I don't think that's what we want.

Reply:
You are right. Corrected